domain/relation/service/relation.go

```diff
@@ -478,11 +538,272 @@ func (s *WatchableService) WatchApplicationSettings(
-	return nil, coreerrors.NotImplemented
+	if err := subordinateAppID.Validate(); err != nil {
+		return nil, errors.Errorf(
+			"%w: %w", relationerrors.ApplicationIDNotValid, err)
+	}
+	currentRelations := make(map[corerelation.UUID]relation.RelationLifeSuspendedData)
+	// returns a set of relation keys if the life or suspended status has changed
+	// for any relation this application is part of.
+	principalAppID, err := s.st.GetPrincipalApplicationID(ctx, subordinateAppID)
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	lifeTableName, suspendedTableName, query := s.st.InitialWatchLifeSuspendedStatus(principalAppID)
+	return s.watcherFactory.NewNamespaceMapperWatcher(
+		func(ctx context.Context, txn database.TxnRunner) ([]string, error) {
+			relationUUIDStrings, err := query(ctx, txn)
+			if err != nil {
+				return nil, errors.Capture(err)
+			}
+
+			initialResults := make([]string, len(relationUUIDStrings))
+
+			for i, relUUID := range relationUUIDStrings {
+				relUUID := corerelation.UUID(relUUID)
+				relationData, err := s.st.WatchLifeSuspendedStatusMapperData(ctx, relUUID, principalAppID)
+				if errors.Is(err, relationerrors.ApplicationNotFoundForRelation) {
+					continue
+				} else if err != nil {
+					return nil, errors.Capture(err)
+				}
+				currentRelations[relUUID] = relationData
+				initialResults[i] = relationData.Key.String()
+			}
+
+			return initialResults, nil
+		},
+		func(ctx context.Context, txn database.TxnRunner, changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+			// If there are no changes, return no changes.
+			if len(changes) == 0 {
+				return nil, nil
+			}
+			var err error
+			changeEvents := []changestream.ChangeEvent{}
+			changeEvents, currentRelations, err = s.changeEventsForPrincipalLifeSuspendedStatusMapper(
+				ctx,
+				subordinateAppID,
+				principalAppID,
+				changeEvents,
+				currentRelations,
+			)
+			if err != nil {
+				return nil, errors.Capture(err)
+			}
+			return changeEvents, nil
+		},
+		eventsource.NamespaceFilter(lifeTableName, changestream.All),
+		eventsource.NamespaceFilter(suspendedTableName, changestream.All),
+	)
+}
+
+func (s *WatchableService) changeEventsForPrincipalLifeSuspendedStatusMapper(
+	ctx context.Context,
+	principalAppID application.ID,
+	subordinateAppID application.ID,
+	changes []changestream.ChangeEvent,
+	currentRelations map[corerelation.UUID]relation.RelationLifeSuspendedData,
+) ([]changestream.ChangeEvent, map[corerelation.UUID]relation.RelationLifeSuspendedData, error) {
+
+	changeEvents := make([]changestream.ChangeEvent, 0)
+	for _, change := range changes {
+		relUUID := corerelation.UUID(change.Changed())
+
+		changedRelationData, err := s.st.WatchLifeSuspendedStatusMapperData(ctx, relUUID, principalAppID)
+		if errors.Is(err, relationerrors.ApplicationNotFoundForRelation) {
+			continue
+		} else if errors.Is(err, relationerrors.RelationNotFound) {
+			delete(currentRelations, relUUID)
```

Comment:
Don't delete it from the map and return it at the same time. We should copy the relations map and then delete it and then return it.